api/service/networkinfo/service.go

```diff
@@ -132,9 +131,11 @@ func (s *Service) DoService() {
 					if err != nil {
 						continue
 					}
-					nip := netaddr.(*net.TCPAddr).IP.String()
-					if strings.Compare(nip, "127.0.0.1") != 0 {
-						ip = nip
+					nip := netaddr.(*net.TCPAddr).IP
+					if nip.IsInterfaceLocalMulticast() || nip.IsLinkLocalMulticast() || nip.IsLoopback() || nip.IsMulticast() {
```

Comment:
Maybe:
```suggestion
					if nip.GlobalUnicast() || CGNPrefix.HasIP(nip) {
```

and

```go
// CGNPrefix is the carrier-grade NAT address space (RFC 6598).
var CGNPrefix = netaddr.MustNewIPNetwork("100.64.0.0/10")
```

This covers all of the 4 cases plus a few more: Unspecified (0.0.0.0) and global broadcast (255.255.255.255), and CGN.  It is also closer to our original intention: We want nodes to publish only globally-routable unicast address.

Reply:
sure.